### PR TITLE
[Opt](ShortCircuit) opt some serialization and fix error when prepared statment excute failed

### PR DESCRIPTION
**Pull request by eldenmoon**
1. If execute failed in ExecuteCommand run method, then `needAnalyze` should be true, in case that the shortCircuitQueryContext is null
2. Optimization on serialization fields to client
3. The grpcThreadPool should be static to avoid too many thread pools and better metrics view

### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShortCircuitQueryContext.java [hunk 1]
```diff
@@ -57,8 +56,28 @@ public class ShortCircuitQueryContext {
     public final Queriable analzyedQuery;
     // Serialized mysql Field, this could avoid serialize mysql field each time sendFields.
     // Since, serialize fields is too heavy when table is wide
-    public Map<String, byte[]> serializedFields =  Maps.newHashMap();
+    List<byte[]> serializedFields = new ArrayList();
```
Comment (xiaokang) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShortCircuitQueryContext.java:59]: use map<Int, byte[]>

Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ShortCircuitQueryContext.java]: done

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]

**Pull request merged**